### DISCUSSION TITLE
How can I see warnings only after saving the file?

### DISCUSSION THREAD

**Discussion by adamckirst** [neoclide/coc.nvim#4356] [Q&A] [answered]
[Noob alert]
I've just started using CoC and it annoys me a lot to see warning of things I am already aware of.
An example would be "Non-void function does not return a value". I am writing the function and will get there eventually, so I don't need to see the error.
Is there a way of CoC only checking for errors after writing the file with ":w"?
Thank you!
(I've tried using RefreshAfterSave but I don't think Diagonsis means what I mentioned here)

**Comment by fannheyward** [comment, marked as answer]
Try `pullDiagnostic.onChange` to `false` and `pullDiagnostic.onSave` to `true`.

But this needs language server supports this. From LSP defined, language server publish diagnostics to client, client can't pull diagnostics from time to time. From LSP 3.17, you can do `pullDiagnostic`, but not all language server support this feature.

**Reply by adamckirst** [reply]
Great! Thank you so much for the instructions!

**Comment by chemzqm** [comment]
See `:h coc-pullDiagnostics`

**Reply by adamckirst** [reply]
Thank you! Now it's the perfect plugin
